Userland/Libraries/LibJS/Runtime/PropertyName.cpp

```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright (c) 2021, the SerenityOS developers.
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include "PropertyName.h"
+#include "AbstractOperations.h"
+#include "GlobalObject.h"
+
+namespace JS {
+
+Value PropertyName::to_canonical_numeric_index_string(GlobalObject& global_object) const
+{
+    VERIFY(is_number() || is_string());
+
+    if (is_number())
+        return Value(m_number);
+
+    return canonical_numeric_index_string(global_object, js_string(global_object.vm(), m_string));
+}
```

Comment:
id argue for putting this in the header BTW, since its smaller than the other header methods, and will allow inlining. but otherwise LGTM

Reply:
Since we use `global_object.vm()` I could not put it in the header since it caused dependency cycles.
But doesn't matter anymore since I'll move it to AbstractOperations